crypto/ml_kem/ml_kem-c.inc

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Copyright (c) 2024, Google Inc. */
+
+#include <stdlib.h>
+#include <openssl/e_os2.h>
+#include <openssl/rand.h>
+#include <internal/sha3.h>
+#include <internal/constant_time.h>
+#include <internal/common.h>
+#include <crypto/ml_kem.h>
+#include "ml_kem_local.h"
+#ifndef NDEBUG
+# include <stdio.h>
+#endif
+
+/*
+ * Variant-independence macros, variously shortened for convenience and
+ * internal consumption by original BoringSSL code.
+ */
+# if BITS == 512
+# define kemsym(a, b)       a##_512_##b
```

Comment:
Are macro names supposed to be uppercase?

Reply:
These are not used outside their source file, difference considerations apply.
Also only used to alias some symbols at the top of the file, and actually used in code.
```
29:# define kemsym(a, b)       a##_512_##b
31:# define kemsym(a, b)       a##_768_##b
33:# define kemsym(a, b)       a##_1024_##b
40:# define RANK           kemsym(ML_KEM,RANK)
41:# define ETA1           kemsym(ML_KEM,ETA1)
42:# define ETA2           kemsym(ML_KEM,ETA2)
43:# define DU             kemsym(ML_KEM,DU)
44:# define DV             kemsym(ML_KEM,DV)
45:# define SECBITS        kemsym(ML_KEM,RNGSEC)
70:typedef kemsym(ossl_ml_kem,vector)      vector;
71:typedef kemsym(ossl_ml_kem,public_key)  public_key;
72:typedef kemsym(ossl_ml_kem,private_key) private_key;
75:# define genkey_rand                kemsym(ossl_ml_kem,genkey_rand)
76:# define encap_rand                 kemsym(ossl_ml_kem,encap_rand)
77:# define decap                      kemsym(ossl_ml_kem,decap)
80:# define genkey_seed                kemsym(ossl_ml_kem,genkey_seed)
81:# define encap_seed                 kemsym(ossl_ml_kem,encap_seed)
82:# define parse_public_key           kemsym(ossl_ml_kem,parse_public_key)
83:# define encode_public_key          kemsym(ossl_ml_kem,encode_public_key)
84:# define parse_private_key          kemsym(ossl_ml_kem,parse_private_key)
85:# define encode_private_key         kemsym(ossl_ml_kem,encode_private_key)
86:# define get_vinfo                  kemsym(ossl_ml_kem,get_vinfo)
```
Actual macros used in the code are upper case. And FWIW, though not relevant here, function-like macros that use each argument exactly once are sometimes OK in lower-case, unless it is unclear whether one can really treat such a macro as a function and try to take its address (e.g. pass its name as an argument).

Reply:
See (https://openssl-library.org/policies/technical/coding-style)

Reply:
Maybe leave it if we are potentially going to rework this at some point..

Reply:
Leaving it be. There's no text about macros redefining symbols for linkage. But closest (since the new symbol is still a function) is:
```
Macro names should be in uppercase, but macros resembling functions may be written in lower case.
```